esiclient/utils.py

```diff
@@ -25,33 +25,39 @@ def get_network_display_name(network):
         return network.name
 
 
-def get_network_info_from_port(port, client):
+def get_network_info_from_port(port, client, networks_dict=None):
     """Return Neutron network name and ips from port
 
     :param port: a Neutron port
+    :param networks_dict: networks dict {id:network}
     :param client: neutron client
     """
-    network = client.get_network(port.network_id)
+    if networks_dict:
+        network = networks_dict.get(port.network_id)
+    else:
+        network = client.get_network(port.network_id)
```

Comment:
I think it may be safer to have networks_dict default to {}, and to have the conditional be `if the networks_dict contains the network_id as a key`.

Reply:
Done!